tools/topology/topology2/get_abi.sh

```diff
@@ -11,6 +11,29 @@ cat <<EOF_HEADER
 Object.Base.manifest."sof_manifest" {
 	Object.Base.data."SOF ABI" {
 EOF_HEADER
-printf '\t\tbytes\t"0x%02x,' "$ABI_MAJOR"
-printf "0x%02x," "$ABI_MINOR"
-printf '0x%02x\"\n\t}\n}' "$ABI_PATCH"
+
+print_1byte()
+{
+	printf '0x%02x' "$(("$1" & 0xff))"
+}
+
+print_2bytes()
+{
+	printf '0x%02x,0x%02x' "$(("$1" & 0xff))" "$(($(("$1" & 0xff00)) >> 8))"
+}
+
+printf '\t\tbytes\t\"'
+for vers in $ABI_MAJOR $ABI_MINOR; do
+	case "$2" in
+		ipc3) print_1byte "$vers" ;;
+		ipc4) print_2bytes "$vers" ;;
+	esac
+	printf ','
+done
+
+case "$2" in
+	ipc3) print_1byte "$ABI_PATCH" ;;
+	ipc4) print_2bytes "$ABI_PATCH" ;;
+esac
```

Comment:
I don't think you need to special case the last one. I just tried a trailing comma like this and I found no difference in any  .tplg file output:
```
Object.Base.manifest."sof_manifest" {
        Object.Base.data."SOF ABI" {
                bytes   "0x03,0x15,0x00,"
        }
}
```

Reply:
you are testing with building tplg2? BTW this change will need an alsautils change that is not upstream. With that the trailing comma causes an error like this:

saving nhlt as binary in nhlt-cavs-tgl-nocodec.bin
error in nhlt processing
ALSA lib data.c:479:(tplg_parse_data_hex) malformed hex variable list 0x03,0x00,0x15,0x00,0x00,0x00,
ALSA lib data.c:1375:(tplg_parse_data) failed to parse data bytes
ALSA lib parser.c:212:(tplg_load_config) failed to parse topology

Reply:
My bad, I was testing with an older ALSA version, not building tplg2. The tplg2.conf files were generated (that confused me) but not the v2 .tplg files.